src/hivpy/hiv_testing.py

```diff
@@ -49,163 +55,240 @@ def __init__(self, **kwargs):
-                                                   (col.LAST_TEST_DATE, op.eq, None)]
-                                                  ])
-
-            # first time testers
-            untested_population = pop.apply_bool_mask(~pop.get_variable(col.EVER_TESTED, testing_population),
-                                                      testing_population)
-            # repeat testers
-            prev_tested_population = pop.apply_bool_mask(pop.get_variable(col.EVER_TESTED, testing_population),
-                                                         testing_population)
-
-            if len(untested_population) > 0:
-                # test first time testers
-                tested = pop.transform_group([col.EVER_TESTED, col.NP_LAST_TEST, col.NSTP_LAST_TEST],
-                                             self.calc_testing_outcomes,
-                                             sub_pop=untested_population)
+        # apply testing to marked population
+        marked_population = pop.get_sub_pop([(col.TEST_MARK, op.eq, True)])
+        self.apply_test_outcomes_to_sub_pop(pop, marked_population)
+
+    def test_mark_hiv_symptomatic(self, pop):
+        """
+        Mark HIV symptomatic individuals to undergo testing this time step.
+
+        Note: If the simulation is started after date_targeted_testing_plateau
+        (or if date_start_testing is set to be after date_targeted_testing_plateau)
+        the HIV symptomatic testing probabilities will not be updated
+        and this function may not work as expected.
+        """
+        # testing occurs after a certain year
+        if (pop.date.year > self.date_start_testing):
+
+            # update symptomatic test probabilities
+            if pop.date.year <= self.date_targeted_testing_plateau:
+                self.prob_test_who4 = min(0.9, self.prob_test_who4 * self.incr_test_rate_sympt)
+                self.prob_test_tb = min(0.8, self.prob_test_tb * self.incr_test_rate_sympt)
+                self.prob_test_non_tb_who3 = min(0.7, self.prob_test_non_tb_who3 * self.incr_test_rate_sympt)
+
+            # undiagnosed population not scheduled for testing
+            not_diag_tested_pop = pop.get_sub_pop([(col.HIV_DIAGNOSED, op.eq, False),
+                                                   (col.TEST_MARK, op.eq, False)])
+
+            if len(not_diag_tested_pop) > 0:
+                # mark people for testing
+                marked = pop.transform_group([pop.get_correct_column(col.ADC, dt=1),
+                                              pop.get_correct_column(col.TB, dt=1),
+                                              pop.get_correct_column(col.TB, dt=2),
+                                              pop.get_correct_column(col.NON_TB_WHO3, dt=1)],
+                                             self.calc_symptomatic_testing_outcomes,
+                                             sub_pop=not_diag_tested_pop)
```

Comment:
I know I'm risking dragging this PR out further, but I do want to check this behaviour: we're looking at the presence of TB in the last two timesteps, but how does this translate to a different timestep e.g. 1 month? Does this need to be changed? @andrew-phillips-1

Reply:
So do we only want to have a chance of testing based on TB in the first time step after acquiring TB, and then not for subsequent timesteps? Or, if we have a one month timestep, should we have some possibility of being testing throughout that six month period?

Reply:
I think it's enough to test once only for each TB episode in a person without diagnosed HIV